AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2049,6 +2052,67 @@ - (BOOL)supportsRangeManagedInterfaceState
   return ASHierarchyStateIncludesRangeManaged(_hierarchyState);
 }
 
+- (id<ASTraitCollection>)traitCollection
+{
+  ASDN::MutexLocker l(_propertyLock);
+  if (_traitCollection == nil) {
+    _traitCollection = [[ASDisplayTraitCollection alloc] initWithTraits:_internalTraits];
+  }
+  return _traitCollection;
+}
+
+- (void)setTraitCollection:(id<ASTraitCollection>)traitCollection
```

Comment:
We just need some sort of entry point that UIView hosts can pass into the node hierarchy. The responder chain is difficult to add into the picture here as it requires the node to rely upon the state of the view hierarchy. Ideally, we should design this such that the node is unaware of the view hierarchy and acts pretty self-contained once the trait collection has been received by its issuer.